Remove Global Styles from Body.js

SudharakaP opened the pull request:
Resolves https://github.com/opencollective/opencollective/issues/3757

# Screenshots

The following components have been migrated and the screenshots are given below (in the given order); `CreateEvent.js`, `createUpdate.js`, `Hosts.js`, `IncognitoUserCollective.js`, `order.js`, `update.js`, `updates.js`, `editcollective/index.js` and `CreatOrganization.js`

<details>
<summary>Desktop</summary>

![image](https://user-images.githubusercontent.com/12435965/101681884-b0cfcf80-3a17-11eb-8ac5-903c4a2042df.png)

![image](https://user-images.githubusercontent.com/12435965/101682005-e2489b00-3a17-11eb-9e2b-eaaa4d6161d6.png)

![image](https://user-images.githubusercontent.com/12435965/101682186-26d43680-3a18-11eb-846c-f34c6a46c339.png)

![image](https://user-images.githubusercontent.com/12435965/101603999-34a4a000-39b5-11eb-9b39-d8c34d2f4050.png)

![image](https://user-images.githubusercontent.com/12435965/101703847-4713ed80-3a38-11eb-9d11-f67720b9e195.png)

![image](https://user-images.githubusercontent.com/12435965/101703924-76c2f580-3a38-11eb-932e-e1b09129f3ed.png)

![image](https://user-images.githubusercontent.com/12435965/101704018-af62cf00-3a38-11eb-93dd-d10de58080db.png)

![image](https://user-images.githubusercontent.com/12435965/101704109-e1743100-3a38-11eb-9a19-68057f5b0cda.png)

![image](https://user-images.githubusercontent.com/12435965/101704253-29935380-3a39-11eb-85e2-49849046bf1d.png)
</details>

<details>
<summary>Mobile</summary>

![Screen Shot 2020-12-09 at 12 11 55](https://user-images.githubusercontent.com/12435965/101681916-bfb68200-3a17-11eb-8d90-2def642d0348.png)

![Screen Shot 2020-12-09 at 12 13 16](https://user-images.githubusercontent.com/12435965/101682045-f096b700-3a17-11eb-8f2b-646fa27886d6.png)

![Screen Shot 2020-12-09 at 12 16 07](https://user-images.githubusercontent.com/12435965/101682344-54b97b00-3a18-11eb-8f77-9aa5267ecfca.png)

![Screen Shot 2020-12-09 at 00 26 49](https://user-images.githubusercontent.com/12435965/101604040-41c18f00-39b5-11eb-93f1-1430f2fcf906.png)

![Screen Shot 2020-12-09 at 16 05 18](https://user-images.githubusercontent.com/12435965/101703885-5c891780-3a38-11eb-8807-b1e679561c80.png)

![Screen Shot 2020-12-09 at 16 06 50](https://user-images.githubusercontent.com/12435965/101703962-91956a00-3a38-11eb-9512-f1c7169d48d9.png)

![Screen Shot 2020-12-09 at 16 08 02](https://user-images.githubusercontent.com/12435965/101704047-bab5fa80-3a38-11eb-9197-1118724fa886.png)

![Screen Shot 2020-12-09 at 16 09 27](https://user-images.githubusercontent.com/12435965/101704130-ecc75c80-3a38-11eb-949c-972b8b9b5a96.png)

![Screen Shot 2020-12-09 at 16 11 27](https://user-images.githubusercontent.com/12435965/101704276-34e67f00-3a39-11eb-8ca6-0da3cfc224be.png)
</details>

Betree commented on public/static/styles/app.css:
```diff
@@ -270,3 +270,26 @@ a {
 #nprogress .spinner {
   display: none;
 }
+
```
```suggestion
/** @deprecated These legacy styles from `Body.js` should be removed */

```

Betree commented on components/Body.js:
```diff
@@ -1,125 +1,18 @@
   static defaultProps = {
-    withoutGlobalStyles: false,
-    withBorderTop: true,
+    withoutGlobalStyles: true,
```
Since we're not using it anymore, we should remove `withoutGlobalStyles` from the props entirely. We should also remove it from where it's used (ie. `<Body withoutGlobalStyles={false} />`)

Betree reviewed (commented):
Looking good! There's a small change on the navbar, on most pages it seems to be missing the grey border it had below: 

**Before**
![image](https://user-images.githubusercontent.com/1556356/102867219-ba563000-4438-11eb-908b-fab1497b557b.png)

**After**
![image](https://user-images.githubusercontent.com/1556356/102867230-c0e4a780-4438-11eb-9a87-0e190db2ce76.png)

SudharakaP commented:
@Betree : Thanks for the code review. I've done all the changes. For the grey border I have removed a default style from `Body.js` thinking that it's not used at all (it's only used within the file itself; the prop `withBorderTop` isn't used it seems) which I added back. 😄

Betree reviewed (approved):
Nice! I didn't find any other regression, I think this is safe to merge.

Like always, great work on this @SudharakaP :+1:

SudharakaP commented:
> Nice! I didn't find any other regression, I think this is safe to merge.
> 
> Like always, great work on this @SudharakaP +1

You are most welcome. 🤗
